src/WalkingModule/src/Module.cpp

```diff
@@ -1314,3 +1640,287 @@ bool WalkingModule::stopWalking()
+        m_outputStepAdaptation.currentFootRightTwist=m_currentFootRightTwist;
+        m_outputStepAdaptation.adaptedFootLeftTransform=m_adaptatedFootLeftTransform;
+        m_outputStepAdaptation.currentFootLeftTransform=m_currentFootLeftTransform;
+        m_outputStepAdaptation.adaptedFootRightTransform=m_adaptatedFootRightTransform;
+        m_outputStepAdaptation.currentFootRightTransform=m_currentFootRightTransform;
+        m_outputStepAdaptation.adaptedFootLeftAcceleration=m_adaptatedFootLeftAcceleration;
+        m_outputStepAdaptation.currentFootLeftAcceleration=m_currentFootLeftAcceleration;
+        m_outputStepAdaptation.timeIndexAfterPushDetection=m_timeIndexAfterPushDetection;
+        m_outputStepAdaptation.adaptedFootRightAcceleration=m_adaptatedFootRightAcceleration;
+        m_outputStepAdaptation.currentFootRightAcceleration=m_currentFootRightAcceleration;
+        m_outputStepAdaptation.FootTimeIndexAfterPushDetection=m_FootTimeIndexAfterPushDetection;
+
+        m_inputStepAdaptation.dT=m_dT;
+        m_inputStepAdaptation.omega=omega;
+        m_inputStepAdaptation.time=m_time;
+        m_inputStepAdaptation.timeOffset=m_timeOffset;
+        m_inputStepAdaptation.leftStepList=m_jLeftstepList;
+        m_inputStepAdaptation.leftInContact=m_leftInContact;
+        m_inputStepAdaptation.leftFootprints=m_jleftFootprints;
+        m_inputStepAdaptation.rightStepList=m_jRightstepList;
+        m_inputStepAdaptation.rightInContact=m_rightInContact;
+        m_inputStepAdaptation.rightFootprints=m_jRightFootprints;
+        m_inputStepAdaptation.dcmSubTrajectories=m_DCMSubTrajectories;
+        m_inputStepAdaptation.dcmPositionSmoothed=m_DCMPositionSmoothed;
+
+        if(!m_stepAdapter->runStepAdaptation(m_inputStepAdaptation,m_outputStepAdaptation))
+        {
+            yError() << "[WalkingModule::runStepAdaptation] unable to run step adaptation";
+            return false;
+        }
+
+        // adapted dcm trajectory
+        // add the offset on the zmp evaluated by the step adjustment for each footprint in the trajectory
+        // the same approach is used also for the impact time since the step adjustment change the impact time
+        if (!m_inputStepAdaptation.leftInContact.front() || !m_inputStepAdaptation.rightInContact.front())
+        {
+        iDynTree::Vector2 adaptedZMPOffset;
+        iDynTree::toEigen(adaptedZMPOffset) = iDynTree::toEigen(m_stepAdapter->getDesiredZmp()) - iDynTree::toEigen(m_outputStepAdaptation.zmpNominal);
+        double adaptedTimeOffset;
+        int numberOfSubTrajectories = m_inputStepAdaptation.dcmSubTrajectories.size();
+        if (numberOfSubTrajectories<2)
+        {
+            yError() << "[WalkingModule::runningStepAdaptation] the size of dcm subtrajectories should be equal or greater than 2";
+            return false;
+        }
+            auto firstSS = m_inputStepAdaptation.dcmSubTrajectories[numberOfSubTrajectories-2];
```

Comment:
The indentation on this line is the only issue I noticed! Well done @MiladShafiee

Reply:
Actually after the latest modifications it seems there sometimes there is some strange behavior in the simulation that I'm trying to find it! But until now I wasn't able to find! :D

Reply:
After struggling I realize that QT was not building the project completely and was doing something strange and didn't give me any error and I was happy that everything is OK!!!!Now I'm compiling again in terminal an there is some brand new errors!